### PR TITLE
Remove warning messages from `tests/integration_tests/test_lightgbm.py`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

A part of #3815 

## Description of the changes

Fix the test code to remove warning messages from sklearn and lightgbm.

### CODE REVIEW

File: tests/integration_tests/test_lightgbm.py [hunk 1]
```diff
@@ -160,14 +160,15 @@ def objective(
     cv: bool = False,
 ) -> float:
 
-    dtrain = lgb.Dataset(np.asarray([[1.0], [2.0], [3.0]]), label=[1.0, 0.0, 1.0])
+    dtrain = lgb.Dataset(np.asarray([[1.0], [2.0], [3.0], [4.0]]), label=[1.0, 0.0, 1.0, 0.0])
```
Comment (nzw0301) [inline tests/integration_tests/test_lightgbm.py:163]: To resolve the following warning, we need to add one more sample labelled `0`.

```
  /opt/homebrew/Caskroom/miniconda/base/envs/lightgbm/lib/python3.9/site-packages/sklearn/model_selection/_split.py:680: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=2.
    warnings.warn(
```

File: tests/integration_tests/test_lightgbm.py [hunk 2]
```diff
@@ -176,9 +177,8 @@ def objective(
             {"objective": "binary", "metric": ["auc", "binary_error"]},
             dtrain,
             num_boost_round,
-            verbose_eval=False,
             nfold=2,
-            callbacks=[pruning_callback],
+            callbacks=[verbose_callback, pruning_callback],
```
Comment (nzw0301) [inline tests/integration_tests/test_lightgbm.py:181]: `verbose_eval`  has been deprecated as in the following warning message by LightGBM. 
```bash
tests/integration_tests/test_lightgbm.py: 14 warnings
  /opt/homebrew/Caskroom/miniconda/base/envs/lightgbm/lib/python3.9/site-packages/lightgbm/engine.py:620: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
    _log_warning("'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. "
```

**Review by HideakiImamura — APPROVED** [review]
Looks great.

**Comment by HideakiImamura** [comment]
@xadrianzetx Could you review this PR if you have time?

**Review by xadrianzetx — APPROVED** [review]
LGTM!

**Comment by c-bata** [comment]
@xadrianzetx Thank you for your review!
Let me merge this PR since we have two approvals.

**Pull request merged**